src/ATen/native/xpu/mkl/SpectralOps.cpp

```diff
@@ -134,10 +126,10 @@ void _fft_with_size(
   // real/imag dimension must aligned when viewed as of complex type
 
   if (complex_input) {
-    bool need_contiguous = input_.stride(-1) != 1;
-
+    const auto strides = input_.strides();
+    bool need_contiguous = strides.back() != 1;
     for (int64_t i = 0; !need_contiguous && i <= signal_ndim; i++) {
-      need_contiguous |= input_.stride(i) % 2 != 0;
+      need_contiguous |= strides[i] % 2;
```

Comment:
I think you can invoke `contiguous()` anyway. Because `contiguous` will check if the tensor is contiguous or not. Meanwhile, the `need_contiguous |= strides[i] % 2` requires that the stride must be 2-element aligned. Why?

```suggestion
    input_ = input_.contiguous();
```